### PR TITLE
[ModuleInterface] Add mechanism to exclude experimental flags from th…

### PR DESCRIPTION
**Pull request by drexin**
…e module interface

rdar://109722548

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Review by tshortli — APPROVED** [review]

**Comment by drexin** [comment]
@swift-ci smoke test

File: include/swift/Basic/Features.def [hunk 1]
```diff
@@ -57,7 +57,7 @@
 #ifndef EXPERIMENTAL_FEATURE
 // Warning: setting `AvailableInProd` to `true` on a feature means that the flag
 //          cannot be dropped in the future.
-#  define EXPERIMENTAL_FEATURE(FeatureName, AvailableInProd)  \
+#  define EXPERIMENTAL_FEATURE(FeatureName, AvailableInProd, IncludeInModuleInterface) \
```
[the remark this answers is not in the document]
Comment (drexin) [reply include/swift/Basic/Features.def]: Thanks, I changed it to a separate define that falls back to `EXPERIMENTAL_FEATURE` if not defined.

**Comment by drexin** [comment]
@swift-ci smoke test

File: include/swift/Basic/Features.def (hunk 1 above)
[the remark this answers is not in the document]
Comment (tshortli) [reply include/swift/Basic/Features.def]: Cool, LGTM!

**Review by DougGregor — APPROVED** [review]
Works for me.

**Pull request merged**